### ISSUE TITLE
Explain utilities mass balance functions

### ISSUE THREAD

**Issue by msevestre** [Open-Systems-Pharmacology/OSPSuite.ReportingEngine#59]
@abdullahhamadeh Could you explain what those functions are doing? They might be redundant with methods that we have in Core. You can encapsulate them but we should not reimplement (see below)
I'd strongly recommend to use this instead of manipulating the path <= THIS IS NOT SAFE as it implies internal knowledge of how we represent paths

getCompoundsList ,
getPathsForMoleculeAmount 

I don't understand what this guy is doing
envList2PathStringsVector 

Path should never be manipulated by hand. We have some methods (toPathArray and toPathString) that should be used instead of usign a string concat with |

Also comments mentioned environment. What does that mean?

Methods in Simulation

There are a bunch of methods in a simuation to retrieve all molecules (either endogenous such as Enzyme oder Xenobiotic such as molecules)

```R
 #' @description
    #' Returns the name of all endogenous stationary molecules defined in the simulation. (e.g. with the flag IsStationary = TRUE)
    #' This is a typically a molecule that is individual specific such as en Enzyme, Protein, Transporter, FcRn etc.
    allEndogenousStationaryMoleculeNames = function() {
      rClr::clrCall(private$.buildConfiguration, "AllPresentEndogenousStationaryMoleculeNames")
    },
    #' @description
    #' Returns the name of all xenobiotoc floating molecules defined in the simulation. (e.g. with the flag IsStationary = FALSE)
    #' This is typically a molecule that is being explicitely simulated such as Compound, Inhibitor, DrugComplex.
    allXenobioticFloatingMoleculeNames = function() {
      rClr::clrCall(private$.buildConfiguration, "AllPresentXenobioticFloatingMoleculeNames")
    },
    #' @description
    #' Returns the name of all stationary molecules defined in the simulation. (e.g. with the flag IsStationary = TRUE)
    allStationaryMoleculeNames = function() {
      rClr::clrCall(private$.buildConfiguration, "AllPresentStationaryMoleculeNames")
    },
    #' @description
    #' Returns the name of all floating molecules defined in the simulation. (e.g. with the flag IsStationary = FALSE)
    allFloatingMoleculeNames = function() {
      rClr::clrCall(private$.buildConfiguration, "AllPresentFloatingMoleculeNames")
    },
```

**Comment by abdullahhamadeh** [comment]
For mass balance, we need to
1) get a list of all compounds in the model and then, for each compound...
2) get the path to all possible outputs that represent "amounts" (or mass) of the compound
3) add those outputs to the simulation object
4) simulate the updated model
5) sum together the amounts from paths retrieved in 2

The functions you're asking about may well be redundant given core and this can all probably be performed more quickly and efficiently, but I think you would first need to export allEndogenousStationaryMoleculeNames etc, right?  Briefly:

getCompoundsList : method to get a list of all compounds for which to do mass balance (used in step 1)

getPathsForMoleculeAmount : get all paths that represent an amount of the molecule in the system (to be summed up for mass balance) (used in step 2)

envList2PathStringsVector : get all the paths needed for mass balance as strings.  The output of this is used to select the columns of the dataframe returned by getOutputValues that need to be summed up.  So this is used in step 5.

**Comment by msevestre** [comment]
>I think you would first need to export allEndogenousStationaryMoleculeNames etc, right? 

Those are public methods of the class `Simulation`. You can call them already

>getCompoundsList ==allXenobioticFloatingMoleculeNames

**Comment by pchelle** [comment]
Fixed since PR #1136 that re-designed mass balance to account for user-defined settings.
Each function and step has been documented appropriately, additionally a vignette has been created to illustrate how to work with mass balance from a user perspective
